src/ensembl/src/content/app/browser/Browser.tsx

```diff
@@ -96,32 +119,64 @@ export const Browser: FunctionComponent<BrowserProps> = (
   };
 
   useEffect(() => {
-    const { stableId } = props.match.params;
-    const location = props.location.search;
+    if (props.match.params.genomeId === undefined) {
+      const locationStr = getChrLocationStr(
+        props.chrLocation[props.activeGenomeId]
+      );
+
+      urlFor.browser(
+        props.activeGenomeId,
+        props.activeEnsObjectId[props.activeGenomeId],
+        locationStr
+      );
+    } else {
+      setTrackStatesFromStorage(browserStorageService.getTrackStates());
+    }
+  }, []);
+
+  useEffect(() => {
+    const { genomeId } = props.match.params;
+
+    props.updateBrowserActiveGenomeIdAndSave(genomeId);
+    props.fetchGenomeTrackCategories(genomeId);
+    props.fetchGenomeInfo(genomeId);
+  }, [props.match.params.genomeId]);
+
+  useEffect(() => {
+    const { focus, location } = props.browserQueryParams;
+    const { genomeId } = props.match.params;
     const chrLocation = getChrLocationFromStr(location);
 
     dispatchBrowserLocation(chrLocation);
-
-    props.fetchEnsObjectData(stableId);
-  }, [props.match.params.stableId]);
+    props.fetchEnsObject(focus, genomeId);
+    props.fetchEnsObjectTracks(focus, genomeId);
+  }, [props.browserQueryParams.focus]);
 
   useEffect(() => {
-    const [, chrStart, chrEnd] = props.chrLocation;
+    const chrLocationForGenome = props.chrLocation[props.activeGenomeId] || [
+      '',
+      0,
+      0
+    ];
+    const [, chrStart, chrEnd] = chrLocationForGenome;
```

Comment:
perhaps slightly clearer and cleaner will be:
```
const chrLocationForGenome = props.chrLocation[props.activeGenomeId] || [];
const [, chrStart = 0, chrEnd = 0] = chrLocationForGenome;
```
_(no need to put the first item in the default array if we don't care about it anyway)_